dabuild.in

```diff
@@ -73,6 +73,26 @@ ABUILD_VOLUMES="-v ${HOME}/.abuild:/home/builder/.abuild \
   -v ${PWD%/aports/*}/aports:/home/builder/aports \
   -v ${ABUILD_PACKAGES}:/home/builder/packages"
 
+## if we have ~/.ccache on host, bind-mount it; else use named per-arch volumes
+if [ -w "$HOME/.ccache" ]; then
+  ccvol=$HOME/.ccache/$DABUILD_ARCH
+  if [ ! \( -d "$ccvol" -a -w "$ccvol" \) ]; then
```

Comment:
What is the reason to test for directory existence after which you check if the file/directory is writable, isn't that exactly what `[ ! -w "$ccvol" ]` will do?
Even more simple is to use mkdir -p which will create the directory if needed or returns 0 if already exists